packages/connection/src/Manager.php

```diff
@@ -1290,23 +1292,23 @@ protected function verify_secrets( $action, $secret_1, $user_id ) {
 		}
 
 		if ( self::SECRETS_MISSING === $stored_secrets ) {
```

Comment:
Because we are now exiting later, rather than returning on the first error, we may end up overwriting the $error value before returning it. For example, if `verify_secret_1_missing` is recorded, then `verify_secrets_expired`, we will only see the later error and not the earlier one. This could make things a little harder to debug in terms of finding the root cause.

I would recommend making this check an elsif like all the ones above it.

Reply:
Change made in 1f7a673.